plugins/vpc-shared-eni/network/bridge_linux.go

```diff
@@ -125,48 +125,78 @@ func (nb *BridgeBuilder) FindOrCreateEndpoint(nw *Network, ep *Endpoint) error {
 	// Connect the bridge to the target network namespace with a veth pair.
 	err = nb.createVethPair(nw.BridgeIndex, targetNetNS, vethLinkName, vethPeerName)
 	if err != nil {
-		log.Errorf("Failed to create veth pair: %v.", err)
+		log.Errorf("Failed to create veth pair %s: %v.", vethLinkName, err)
 		return err
 	}
 
-	gatewayIPAddress := nw.GatewayIPAddress
+	var epIPAddresses []net.IPNet
+	var gatewayIPv4Address net.IP
+	var gatewayIPv6Address net.IP
+	var gatewayIPAddresses []net.IP
 	var gatewayMACAddress net.HardwareAddr
 
-	// Check whether the endpoint is in the same subnet as the ENI.
-	epSubnetPrefix := vpc.GetSubnetPrefix(ep.IPAddress)
-	eniSubnetPrefix := vpc.GetSubnetPrefix(nw.ENIIPAddress)
-	sameSubnet := (epSubnetPrefix.String() == eniSubnetPrefix.String())
-
-	if nw.BridgeType == config.BridgeTypeL3 || !sameSubnet {
-		// Route ingress traffic for the endpoint to the bridge.
-		route := &netlink.Route{
-			LinkIndex: nw.BridgeIndex,
-			Scope:     netlink.SCOPE_LINK,
+	if nw.BridgeType == config.BridgeTypeL3 {
+		// Configure the endpoint to relay the default gateway traffic to the on-link bridge.
+		bridgeLink, err := netlink.LinkByIndex(nw.BridgeIndex)
+		if err == nil {
+			gatewayMACAddress = bridgeLink.Attrs().HardwareAddr
 		}
 
-		// If it is in the same subnet, the IP address must have been allocated from a VPC subnet,
-		// so restrict the route to this IP address only. Otherwise, the IP address is from a CIDR
-		// block delegated to this instance, so add a single route for the entire CIDR block.
-		if sameSubnet {
-			dst := *ep.IPAddress
+		for _, ipAddr := range ep.IPAddresses {
+			dst := ipAddr
```

Comment:
Let's add a comment here:
// Route ingress traffic for this IP address to the bridge.